### PR TITLE
wp-cli 1.5.0 (new formula)

### PR DESCRIPTION
**Pull request by javian**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This is a migration from homebrew-php

### CODE REVIEW

**Comment by josegonzalez** [comment]
@javian we can just print the message with printf or echo, right?

**Comment by javian** [comment]
@josegonzalez yeah, I've changed it back now - it didn't really bring any additional benefits.

**Comment by javian** [comment]
It looks like the current set up cause some issues with the configuration compared to what is shown here https://make.wordpress.org/cli/handbook/installing/ .
```
➜  homebrew-core git:(master) wp --info
OS:	Darwin 17.4.0 Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64
Shell:	/bin/zsh
PHP binary:	/usr/local/Cellar/php/7.2.3/bin/php
PHP version:	7.2.3
php.ini used:	/usr/local/etc/php/7.2/php.ini
WP-CLI root dir:	phar://wp-cli.phar
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
WP-CLI packages dir:
WP-CLI global config:
WP-CLI project config:
WP-CLI version:	1.5.0
```

File: Formula/wp-cli.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class WpCli < Formula
+  desc "Command-line interface for WordPress"
+  homepage "https://wp-cli.org"
+  url "https://github.com/wp-cli/wp-cli/releases/download/v1.5.0/wp-cli-1.5.0.phar"
+  sha256 "f615d57957e66a09f57acc844a1fc5402e9fa581dcb387bbe1affc4d155baf9d"
+  revision 1
```
Comment (SMillerDev) [inline Formula/wp-cli.rb]: Is this needed?

**Review by SMillerDev — APPROVED** [review]

File: Formula/wp-cli.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/wp-cli.rb]: No `revision` please.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/wp-cli.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "f615d57957e66a09f57acc844a1fc5402e9fa581dcb387bbe1affc4d155baf9d"
+
+  bottle :unneeded
+
+  def install
+    bin.install "wp-cli-#{version}.phar" => "wp"
+  end
+
+  test do
+    system "#{bin}/wp", "--info"
```
Comment (fxcoudert) [inline Formula/wp-cli.rb]: Can you include a test that exercises some of the software's functionality? Like get the status of a site, or try an operation without proper credentials and check for failure?

**Review by fxcoudert — CHANGES_REQUESTED** [review]
I'm seeing on the webpage:

> You can update WP-CLI with `wp cli update` (doc), or by repeating the installation steps

This is something that is frowned upon in Homebrew, as it circumvents our update mechanism. Do we have any option to disable it?

File: Formula/wp-cli.rb (hunk 2 above)
Comment (javian) [reply Formula/wp-cli.rb]: Yes I'll do that.

**Comment by javian** [comment]
@fxcoudert for wp-cli the update will fail since it fails to detect that it has write permissions to the file that it wants to overwrite.
```
➜  Formula git:(wpcli150) ✗ wp cli update
Error: /usr/local/Cellar/wp-cli/1.5.0/bin/wp is not writable by current user.
```
This is probably a bug since composer can self update just fine.

**Comment by javian** [comment]
I've added a test now which at least uses functionality other than displaying text. To do more like install a wordpress plugin I would need a mysql database since the cli tool needs to use the wordpress setup to function.

**Comment by javian** [comment]
Why was this PR closed ?

**Comment by fxcoudert** [comment]
Thanks @javian!

**Pull request closed without merging**